packages/react/src/internal/useResizeObserver.ts

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright IBM Corp. 2025, 2025
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { useRef, useState, useLayoutEffect, useEffect } from 'react';
+
+export const useResizeObserver = ({ ref, onResize }) => {
```

Comment:
Can you type the options?

```suggestion
export const useResizeObserver = ({
  ref,
  onResize,
}: {
  ref: RefObject<HTMLElement>;
  onResize?: (rect: DOMRectReadOnly) => void;
}) => {
```